src/engine/Bind.cpp

```diff
@@ -81,104 +81,138 @@ std::vector<QueryExecutionTree*> Bind::getChildren() {
+    //
+    // TODO: In most BIND operations, nothing is added to the local vocabulary,
+    // so it would be more efficient to first share the pointer here (like with
+    // `shareLocalVocabFrom`) and only copy it when a new word is about to be
+    // added. Same for GROUP BY.
+    LocalVocab localVocab = subRes->getCopyOfLocalVocab();
+    IdTable result = applyBind(subRes->idTable(), &localVocab);
+    LOG(DEBUG) << "BIND result computation done." << std::endl;
+    return {std::move(result), resultSortedOn(), std::move(localVocab)};
+  }
+  auto localVocab = std::make_shared<LocalVocab>();
+  auto generator =
+      [](std::shared_ptr<LocalVocab> vocab, auto applyBind,
+         std::shared_ptr<const Result> result) -> cppcoro::generator<IdTable> {
+    for (IdTable& idTable : result->idTables()) {
+      co_yield applyBind(idTable, vocab.get());
+    }
+    std::array<const LocalVocab*, 2> vocabs{vocab.get(), &result->localVocab()};
+    *vocab = LocalVocab::merge(std::span{vocabs});
+  }(localVocab, std::move(applyBind), std::move(subRes));
+  return {std::move(generator), resultSortedOn(), std::move(localVocab)};
+}
+
+// _____________________________________________________________________________
+IdTable Bind::computeExpressionBind(
+    LocalVocab* outputLocalVocab,
+    ad_utility::SimilarTo<IdTable> auto&& inputIdTable,
+    const LocalVocab& inputLocalVocab,
+    sparqlExpression::SparqlExpression* expression,
+    std::optional<std::pair<size_t, size_t>> subrange) const {
   sparqlExpression::EvaluationContext evaluationContext(
-      *getExecutionContext(), _subtree->getVariableColumns(),
-      inputResultTable.idTable(), getExecutionContext()->getAllocator(),
-      inputResultTable.localVocab(), cancellationHandle_, deadline_);
+      *getExecutionContext(), _subtree->getVariableColumns(), inputIdTable,
+      getExecutionContext()->getAllocator(), inputLocalVocab,
+      cancellationHandle_, deadline_);
+
+  if (subrange.has_value()) {
+    auto [beginIndex, endIndex] = subrange.value();
+    evaluationContext._beginIndex = beginIndex;
+    evaluationContext._endIndex = endIndex;
+  }
 
   sparqlExpression::ExpressionResult expressionResult =
       expression->evaluate(&evaluationContext);
 
-  const auto input = inputResultTable.idTable().asStaticView<IN_WIDTH>();
-  auto output = std::move(*outputIdTable).toStatic<OUT_WIDTH>();
-
-  // first initialize the first columns (they remain identical)
-  const auto inSize = input.size();
-  output.reserve(inSize);
-  const auto inCols = input.numColumns();
-  // copy the input to the first numColumns;
-  for (size_t i = 0; i < inSize; ++i) {
-    output.emplace_back();
-    for (size_t j = 0; j < inCols; ++j) {
-      output(i, j) = input(i, j);
-    }
-    checkCancellation();
-  }
+  auto output = subrange.has_value()
+                    ? cloneSubView(inputIdTable, subrange.value())
+                    : std::move(inputIdTable).cloneOrMove();
```

Comment:
```suggestion
                    : AD_FWD(inputIdTable).cloneOrMove();
```
(see below, with those changes you should still get the `move` where you want it, and sonarcloud should be happier.